### PR TITLE
azure - KeyVault integration docs

### PR DESCRIPTION
**Pull request by logachev**

### CODE REVIEW

File: docs/source/azure/configuration/authentication.rst [hunk 1]
```diff
@@ -112,6 +112,32 @@ You must set `AZURE_SUBSCRIPTION_ID` as with the other authentication types.
     AZURE_SUBSCRIPTION_ID=subscriptionId
     AZURE_CLIENT_ID=clientId
 
+Azure Key Vault Integration
+---------------------------
+
+If you run Custodian inside Azure VM, AKS, ACI or Azure Function you can leverage Azure Key Vault to store
+Service Principal credentials. You can store json-formatted authentication file as a Key Vault secret.
```
Comment (aluong) [inline docs/source/azure/configuration/authentication.rst]: (minor) `You can store the json-formatted authentication file as a Key Vault secret.`

File: docs/source/azure/configuration/authentication.rst [hunk 2]
```diff
@@ -112,6 +112,32 @@ You must set `AZURE_SUBSCRIPTION_ID` as with the other authentication types.
...
+It will be used to extend your local authentication options. It is recommended to configure Subscription ID
+and Tenant ID locally and store SP credentials in Azure Key Vault.
```
Comment (aluong) [inline docs/source/azure/configuration/authentication.rst]: It is recommended to configure the **Subscription Id** and **Tenant ID** locally and store the Service Principal credentials in Azure Key Vault.

Highlight the keywords - SubId, TenantId

File: docs/source/azure/configuration/authentication.rst [hunk 3]
```diff
@@ -112,6 +112,32 @@ You must set `AZURE_SUBSCRIPTION_ID` as with the other authentication types.
...
+
+Cloud Custodian leverages Managed Service Identity or User Assigned Identity to access Key Vault and retrieve
+extended configuration.
```
Comment (aluong) [inline docs/source/azure/configuration/authentication.rst]: Missing "... [the] extended configuration."

File: docs/source/azure/configuration/authentication.rst [hunk 4]
```diff
@@ -112,6 +112,32 @@ You must set `AZURE_SUBSCRIPTION_ID` as with the other authentication types.
     AZURE_SUBSCRIPTION_ID=subscriptionId
     AZURE_CLIENT_ID=clientId
 
+Azure Key Vault Integration
+---------------------------
+
+If you run Custodian inside Azure VM, AKS, ACI or Azure Function you can leverage Azure Key Vault to store
```
Comment (stefangordon) [inline docs/source/azure/configuration/authentication.rst]: > or *an* Azure Function


**or**

> or Azure Function*s*

File: docs/source/azure/configuration/authentication.rst [hunk 5]
```diff
@@ -112,6 +112,32 @@ You must set `AZURE_SUBSCRIPTION_ID` as with the other authentication types.
...
+the extended configuration.
+
+Example of the secret stored in a Key Vault:
+
+.. code-block:: json
+
+    {"client_id": "<your-sp-id>", "client_secret": "<your-sp-password>"}
+
+To enable Key Vault integration support please populate ``AZURE_KEYVAULT_SECRET`` environment variable
+with Secret URI. If you use User Assigned Identity, please provide ``AZURE_KEYVAULT_CLIENT_ID`` with
```
Comment (aluong) [inline docs/source/azure/configuration/authentication.rst]: "... environment variable with the Azure KeyVault's secret URI"

Can we provide an example of the URI or steps of how to obtain one.

File: docs/source/azure/configuration/authentication.rst [hunk 6]
```diff
@@ -112,6 +112,32 @@ You must set `AZURE_SUBSCRIPTION_ID` as with the other authentication types.
     AZURE_SUBSCRIPTION_ID=subscriptionId
     AZURE_CLIENT_ID=clientId
 
+Azure Key Vault Integration
+---------------------------
+
+If you run Custodian inside Azure VM, AKS, ACI or Azure Functions you can leverage Azure Key Vault to store
+Service Principal credentials. You can store the json-formatted authentication file as a Key Vault secret.
+It will be used to extend your local authentication options. It is recommended to configure **Subscription ID**
+and **Tenant ID** locally and store SP credentials in Azure Key Vault.
```
Comment (aluong) [inline docs/source/azure/configuration/authentication.rst]: "...and store **the service principal's** credentials in..." (grammer)

**Pull request merged**